crates/ruff_python_parser/src/lexer.rs

```diff
@@ -157,6 +159,7 @@ impl<'source> Lexer<'source> {
             source: input,
             cursor: Cursor::new(input),
             mode,
+            fstring_stack: vec![],
```

Comment:
Nit: Have you considered introducing a `FStringStack` newtype wrapper around `Vec`? Could that reduce some repetitive code inside of the lexer? Similar to `Indentations`

Reply:
We can introduce that abstraction but can you expand on "Could that reduce some repetitive code inside of the lexer?". What part is repetitive?